examples/dual_variable_example/dual_variable_example.py

```diff
@@ -0,0 +1,287 @@
+# -*- coding: utf-8 -*-
+
+"""
+General description
+-------------------
+
+A basic example to show how to get the dual variables from the system. Try
+to understand the plot.
+
+
+Installation requirements
+-------------------------
+
+This example requires the version v0.5.x of oemof. Install by:
+
+    pip install 'oemof.solph>=0.5,<0.6'
+
+Optional:
+
+    pip install matplotlib
+
+SPDX-FileCopyrightText: Uwe Krien <krien@uni-bremen.de>
+
+SPDX-License-Identifier: MIT
+"""
+
+
+# ****************************************************************************
+# ********** PART 1 - Define and optimise the energy system ******************
+# ****************************************************************************
+
+###############################################################################
+# imports
+###############################################################################
+
+import logging
+import os
+import pprint as pp
```

Comment:
```suggestion
import pprint as pp
```
All three imports are unused. However, pprint might be used if matplotlib is really optional. (Maybe, it shouldn't be optional. The plot seems to be essential.)

Reply:
It was my plan to make matplotlib mandatory. With #850 it will be possible to install `pip install oemof.solph[examples]` which will install matplotlib. So I added this already to the header.